.github/scripts/backport.py

```diff
@@ -0,0 +1,479 @@
+
+    for c in sorted(comments, key=comment_ts, reverse=True):
+        body = c.get("body") or ""
+        if marker in body:
+            return comment_ts(c)
+    return None
+
+
+def run_remind(prefetched_prs: List[Dict[str, Any]], pending_label_age_days: int, lookback_days: int) -> int:
+    """Post reminders using prefetched merged PR list."""
+
+    now = dt.datetime.now(dt.timezone.utc)
+    threshold = now - dt.timedelta(days=pending_label_age_days)
+    logger.info(
+        f"prefetched={len(prefetched_prs)} lookback_days={lookback_days} pending_label_age_days={pending_label_age_days} now={now.isoformat()} threshold={threshold.isoformat()}"
+    )
+    for pr in prefetched_prs:
+        try:
+            if not pr:
+                continue
+            info = PRInfo(pr)
+            if not has_pending_label([{"name": l} for l in info.labels]):
+                continue
+            number = info.number
+            author = pr.get("user", {}).get("login", "PR author")
+            comments = get_issue_comments(number)
+            prev_time = last_reminder_time(comments, COMMENT_MARKER_BASE)
+            if prev_time is None:
+                post_comment(number, f"{COMMENT_MARKER_BASE}\n@{author}\n{REMINDER_BODY}")
+                logger.info(f"PR #{number}: initial reminder posted")
+            elif prev_time < threshold:
+                post_comment(number, f"{COMMENT_MARKER_BASE}\n@{author}\n{REMINDER_BODY}")
+                logger.info(f"PR #{number}: follow-up reminder posted (prev {prev_time.isoformat()})")
+            else:
+                logger.info(f"PR #{number}: cooling period not elapsed (prev {prev_time.isoformat()})")
+        except Exception as ex:
+            logger.error(f"Remind error for PR #{pr.get('number', '?')}: {ex}")
+            continue
+    return 0
+
+
+# ----------------------------- CLI -----------------------------
+def is_dry_run() -> bool:
+    return bool(CONFIG.get("dry_run"))
+
+
+def require_mandatory_vars() -> None:
+    """Validate critical environment / CLI inputs using CONFIG."""
+    if not CONFIG.get("token"):
+        raise RuntimeError("Missing BACKPORT_TOKEN from environment.")
+    repo = CONFIG.get("repo")
+    if not repo or not re.match(r"^[^/]+/[^/]+$", str(repo)):
+        raise RuntimeError("Missing or invalid GITHUB_REPOSITORY. Either set it or pass --repo (owner/repo)")
+
+
+def configure(args: argparse.Namespace) -> None:
+    """Populate CONFIG, initialize logging, and validate required inputs.
+
+    This centralizes setup so other entry points (tests, future subcommands)
+    can reuse consistent initialization semantics.
+    """
+    CONFIG["dry_run"] = bool(getattr(args, "dry_run", False))
+    CONFIG["log_level"] = "DEBUG" if getattr(args, "debug", False) else "INFO"
+    CONFIG["command"] = getattr(args, "command", None)
+    if getattr(args, "repo", None):
+        CONFIG["repo"] = args.repo
+    setup_logging(CONFIG["log_level"])
```

Comment:
`CONFIG["log_level"]` is `bool | str | None` but `str` expected in `setup_logging()`. You can use `assert` in the preceding line to address that.